app/frontend/src/components/apiAccess/Step2.vue

```diff
@@ -79,20 +81,50 @@ export default {
   name: 'ApiAccessStep2',
   data() {
     return {
+      appNameRules: [
+        v => !!v || 'Name is required',
+        v =>
+          v.length <= FieldValidations.NAME_MAX_LENGTH ||
+          `Name must be ${FieldValidations.NAME_MAX_LENGTH} characters or less`
+      ],
+      appDescriptionRules: [
+        v => !!v || 'Description is required',
+        v =>
+          v.length <= FieldValidations.DESCRIPTION_MAX_LENGTH ||
+          `Description must be ${FieldValidations.DESCRIPTION_MAX_LENGTH} characters or less`
+      ],
       confirmDialog: false,
       fieldValidations: FieldValidations
     };
   },
   computed: {
-    ...mapGetters('apiAccess', [
-      'acronym',
-      'appName',
-      'appDescription',
-      'environment'
-    ])
+    ...mapGetters('apiAccess', ['acronym', 'environment']),
+    // TODO: 2-way binding to vuex state with v-model (https://vuex.vuejs.org/guide/forms.html), but there's got to be a less verbose way...
+    // Going with this to get features working for now
```

Comment:
This is actually one of the potential patterns we can stick with, except in this case we'd no longer use ...mapGetters(...) and just make sure our computed properties directly get and commit from the store.
Ref: https://ypereirareis.github.io/blog/2017/04/25/vuejs-two-way-data-binding-state-management-vuex-strict-mode/#edit-two-way-computed-property

Reply:
Yeah I've seen it for form access/2 way binding. I'd keep the getters for just read since it's more succinct.
So it's probably the right way as you say, I still don't like the overkill syntax :)
I think this is one of the main complaints about Vuex and I've seen it mentioned there could be some big changes in future version.

Reply:
Yea - feels a bit overkill for syntax I agree, but at least it achieves what we need it to do.